pkg/cmd/cli/cmd/create_route.go

```diff
@@ -0,0 +1,454 @@
+	cmd.Flags().String("port", "", "Name of the service port or number of the container port the route will route traffic to")
+	cmd.Flags().String("expose", "", "Name of the service that the new route is exposing")
+	cmd.MarkFlagRequired("expose")
+	cmd.Flags().String("path", "", "Path that the router watches to route traffic to the service.")
+	cmd.Flags().String("cert", "", "Path to a certificate file.")
+	cmd.Flags().String("key", "", "Path to a key file.")
+	cmd.Flags().String("ca-cert", "", "Path to a CA certificate file.")
+
+	return cmd
+}
+
+// CreateEdgeRoute implements the behavior to run the create edge route command.
+func CreateEdgeRoute(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args []string) error {
+	oc, kc, err := f.Clients()
+	if err != nil {
+		return err
+	}
+	ns, _, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+	serviceName, err := resolveServiceName(f, kcmdutil.GetFlagString(cmd, "expose"))
+	if err != nil {
+		return err
+	}
+	routeName, err := resolveRouteName(args)
+	if err != nil {
+		return err
+	}
+	route, err := unsecuredRoute(kc, ns, routeName, serviceName, kcmdutil.GetFlagString(cmd, "port"))
+	if err != nil {
+		return err
+	}
+
+	route.Spec.Host = kcmdutil.GetFlagString(cmd, "hostname")
+	route.Spec.Path = kcmdutil.GetFlagString(cmd, "path")
+
+	route.Spec.TLS = new(api.TLSConfig)
+	route.Spec.TLS.Termination = api.TLSTerminationEdge
+	cert, err := fileutil.LoadData(kcmdutil.GetFlagString(cmd, "cert"))
+	if err != nil {
+		return err
+	}
+	route.Spec.TLS.Certificate = string(cert)
+	key, err := fileutil.LoadData(kcmdutil.GetFlagString(cmd, "key"))
+	if err != nil {
+		return err
+	}
+	route.Spec.TLS.Key = string(key)
+	caCert, err := fileutil.LoadData(kcmdutil.GetFlagString(cmd, "ca-cert"))
+	if err != nil {
+		return err
+	}
+	route.Spec.TLS.CACertificate = string(caCert)
+
+	route, err = oc.Routes(ns).Create(route)
+	if err != nil {
+		return err
+	}
+	mapper, typer := f.Object()
+	resourceMapper := &resource.Mapper{
+		ObjectTyper:  typer,
+		RESTMapper:   mapper,
+		ClientMapper: resource.ClientMapperFunc(f.Factory.RESTClient),
+	}
+	info, err := resourceMapper.InfoForObject(route)
+	if err != nil {
+		return err
+	}
+	shortOutput := kcmdutil.GetFlagString(cmd, "output") == "name"
+	kcmdutil.PrintSuccess(mapper, shortOutput, out, info.Mapping.Resource, info.Name, "created")
+	return nil
+}
+
+const (
+	passthroughRouteLong = `
+Create a route that uses passthrough TLS termination
+
+Specify the service that the new route should expose via the --expose flag.`
+
+	passthroughRouteExample = `  # Create a passthrough route named "my-route" that exposes the frontend service.
+  $ %[1]s create route passthrough my-route --expose frontend
+
+  # Create a passthrough route that exposes the frontend service and specify
+  # a hostname. If the route name is omitted, the service name will be re-used.
+  $ %[1]s create route passthrough --expose frontend --hostname www.example.com`
```

Comment:
`--hostname=`?  I find it helpful to be able to separate flag argument from `bool` flag and regular arg